src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,110 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String RIGHT_LOGIN_EXIST = "qazwsx";
+    private static final int RIGHT_AGE = 18;
+    private static final String RIGHT_PASSWORD = "absabs";
+    private static final String LOGIN_DOES_NOT_EXIST = "abrakadabra";
+    private static final int WRONG_AGE = 17;
+    private static final String WRONG_PASSWORD = "12345";
+    private static RegistrationServiceImpl registration;
+    private User rightUserExist;
+    private User wrongUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registration = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        rightUserExist = new User();
+        rightUserExist.setLogin(RIGHT_LOGIN_EXIST);
+        rightUserExist.setAge(RIGHT_AGE);
+        rightUserExist.setPassword(RIGHT_PASSWORD);
+        new StorageDaoImpl().add(rightUserExist);
+
+        wrongUser = new User();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_userExistTest() {
+        assertThrows(RuntimeException.class, () -> registration.register(rightUserExist));
+    }
+
+    @Test
+    void register_userIsRight_And_DoesNotExist() {
+        User rightDoesNotExist = new User();
+        rightDoesNotExist.setLogin(LOGIN_DOES_NOT_EXIST);
+        rightDoesNotExist.setAge(RIGHT_AGE);
+        rightDoesNotExist.setPassword(RIGHT_PASSWORD);
+        assertDoesNotThrow(() -> registration.register(rightDoesNotExist));
+    }
+
+    @Test
+    void userWillBeNull() {
```

Comment:
For this task use such convention: <methodUnderTest>_<state>_<expectedBehavior>; For example, if we are testing the method register with a null user's age the test method name should be register_nullAge_notOk